### PR TITLE
Add libspdm_ec_set_priv_key function and test

### PR DESCRIPTION
**Pull request by Wenxing-hou**
Fix: #1323 

Signed-off-by: Wenxing Hou <wenxing.hou@intel.com>

### CODE REVIEW

File: include/hal/library/cryptlib/cryptlib_ec.h [hunk 1]
```diff
@@ -45,6 +45,21 @@ extern void libspdm_ec_free(void *ec_context);
 extern bool libspdm_ec_set_pub_key(void *ec_context, const uint8_t *public_key,
                                    size_t public_key_size);
 
+/**
+ * Sets the private key component into the established EC context.
+ *
+ *
```
Comment (jyao1) [inline include/hal/library/cryptlib/cryptlib_ec.h:51]: please describe the size requirement. similar the one described in `libspdm_ec_set_pub_key`.

Comment (Wenxing-hou) [reply include/hal/library/cryptlib/cryptlib_ec.h:54]: Thanks. I have added the private key size description.

File: include/hal/library/cryptlib/cryptlib_ec.h [hunk 2]
```diff
@@ -45,6 +45,24 @@ extern void libspdm_ec_free(void *ec_context);
 extern bool libspdm_ec_set_pub_key(void *ec_context, const uint8_t *public_key,
                                    size_t public_key_size);
 
+/**
```
Comment (steven-bellock) [inline include/hal/library/cryptlib/cryptlib_ec.h:48]: If this function will not be called by any functions in `libspdm/library` it needs to be moved to `os_stub/spdm_crypt_ext_lib/cryptlib_ext.h`

Comment (jyao1) [reply include/hal/library/cryptlib/cryptlib_ec.h:48]: @steven-bellock , this is for cryptlib, not spdm_cryptlib.

**Review by jyao1 — APPROVED** [review]

**Comment by Wenxing-hou** [comment]
> Please drop the CI update - [Add test_crypt for CI](https://github.com/DMTF/libspdm/pull/1331/commits/69db8273a109e23c4044b995f376bc406a5e1ae0), because of #1346.
> 
> Adding new CI should be a separate P-R.

Thanks. I have submitted a new PR for adding test_crypt in CI.

File: include/hal/library/cryptlib/cryptlib_ec.h (hunk 2 above)
Comment (steven-bellock) [reply include/hal/library/cryptlib/cryptlib_ec.h:48]: Ok.

**Pull request merged**